adaptive_scheduler/_server_support/database_manager.py

```diff
@@ -200,8 +200,9 @@ def update(self, queue: dict[str, dict[str, str]] | None = None) -> None:
             return
         if queue is None:
             queue = self.scheduler.queue(me_only=True)
+        job_names_in_queue = [x["job_name"] for x in queue.values()]
         failed = self._db.get_all(
-            lambda e: (e.job_id is not None) and (e.job_id not in queue),  # type: ignore[operator]
+            lambda e: e.job_name is not None and e.job_name not in job_names_in_queue,  # type: ignore[operator]
```

Comment:
This is the key change: the previous logic would not catch DB entries that did not have 'job_id' set, which misses jobs that have been launched but not reported started yet